libdnf/rpm/package_query.cpp

```diff
@@ -2453,6 +2456,56 @@ void PackageQuery::filter_recent(const time_t timestamp) {
     }
 }
 
+void PackageQuery::filter_unneeded() {
+    auto & pool = get_rpm_pool(p_impl->base);
+
+    // get a list of all installed packages
+    auto * installed_repo = pool->installed;
+    if (installed_repo == nullptr) {
+        (*p_impl).clear();
+        return;
+    }
+
+    libdnf::solv::SolvMap installed_solv_map(pool.get_nsolvables());
+    for (int it = installed_repo->start; it < installed_repo->end; ++it) {
+        installed_solv_map.add(it);
```

Comment:
This construct could add non rpm elements to map or not system packages to map - the repository in principle could be interrupted - in future it could create a problems, but not right now.

Reply:
That's a good point, thank you!